### ISSUE TITLE
Error/Warning on alias name clash

### ISSUE THREAD

**Issue by DarkTrick** [gohugoio/hugo#11742]
If the same alias is used for multiple ~~sites~~ pages, throw an error or a warning during generation.

### Example:

*post1.md*
```
+++
title = 'post1'
aliases = ['/111']
+++
post1
```

*post2md*
```
+++
title = 'post2'
aliases = ['/111']
+++
post2
```

**Current behavior**
Silently ignore the fact.

*current* = `hugo v0.120.4-f11bca5fec2ebb3a02727fb2a5cfb08da96fd9df+extended linux/amd64 BuildDate=2023-11-08T11:18:07Z VendorInfo=snap:0.120.4`

**Suggested Behavior**
This should result in an error/warning, that two ~~sites~~ pages use the same aliases.

**Comment by jmooring** [comment]
Until this is implemented, do something like this from your baseof.html...

```
{{ if .IsHome }}
  {{ partial "init.html" . }}
{{ end }}
```

<details>
<summary>layouts/partials/init.html</summary>

```text
{{ $aliases := dict }}
{{ range $p := site.Pages }}
  {{ if .File }}
    {{ range .Aliases }}
      {{ $targets := slice }}
      {{ with index $aliases . }}
        {{ $targets = append $p.File.Path . }}
      {{ else }}
        {{ $targets = slice $p.File.Path }}
      {{ end }}
      {{ $aliases = merge $aliases (dict . $targets) }}
    {{ end }}
  {{ end }}
{{ end }}

{{ range $alias, $targets := $aliases }}
  {{ if gt (len $targets) 1 }}
    {{ errorf "Duplicate aliases: the following pages are aliased to %s: %s" $alias (delimit $targets ", ") }}
  {{ end }}
{{ end }}
```
</details>
<br>

**Comment by philwinder** [comment]
I've only just realised that a lot of my aliases weren't rendering properly and the script above helped me to figure it out. Thanks @jmooring!

I had mistakenly included something like:

```yaml
aliases:
- ""
```

in my archetype and forgot to remove it for some pages. This seemed to cause many pages not render their aliases properly.

Once I included the [script above](https://github.com/gohugoio/hugo/issues/11742#issuecomment-1826871517) and removed the duplicate aliases, all of the aliases started working again.

I'm not sure if my alias rendering issue was caused by the `- ""` alias (which I presume is invalid) or if it was because they were duplicate aliases. Either way, the above check helped me to find the issue.
